### PR TITLE
Fix redirections on empty teams

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Fix redirections on empty teams for guests users

#### Ticket Link
[MM-20152](https://mattermost.atlassian.net/browse/MM-20152)

### CODE REVIEW

File: actions/global_actions.jsx [hunk 1]
```diff
@@ -291,47 +291,61 @@ export async function redirectUserToDefaultTeam() {
...
+        const myMember = getMyTeamMember(state, myTeam.id);
+        if (!myMember) {
+            continue;
         }
 
-        let channelName = LocalStorageStore.getPreviousChannelName(user.id, team.id);
-        let channel = getChannelByName(state, channelName);
-        const channelMember = getMyChannelMember(state, channelName);
+        if (user.id && myTeam) {
+            if (user && Utils.isGuest(user)) {
```
Comment (streamer45) [inline actions/global_actions.jsx]: Why are we checking `user.id` first and then `user`? We already assume those are defined here:

https://github.com/mattermost/mattermost-webapp/blob/63bc1640e66fc895d863f4d32c8e3f6955942e61/actions/global_actions.jsx#L292

And if it's possible for `user` to be `undefined` we should return before that statement.

Also we assume `myTeam` to be defined here:

https://github.com/mattermost/mattermost-webapp/blob/63bc1640e66fc895d863f4d32c8e3f6955942e61/actions/global_actions.jsx#L307

Comment (jespino) [reply actions/global_actions.jsx]: Thanks, I realized now that if you don't have user, makes no sense to keep going, so I'll ignore it, and all the `myTeams` members should be a valid object, so we can remove the check.

**Comment by jespino** [comment]
@streamer45 PTAL

**Comment by jespino** [comment]
@srkgupta Do you have redirections tests in Cypress, right? Would be great to execute them before we merge it.

**Comment by srkgupta** [comment]
There are also 3 unit tests failed which are related to redirections. Can you also ensure that those are working.

**Comment by jespino** [comment]
@srkgupta yes, I was fixing ling errors and by the way I broke the tests and the behavior.

**Comment by jespino** [comment]
@srkgupta Now should be fixed

**Pull request merged**